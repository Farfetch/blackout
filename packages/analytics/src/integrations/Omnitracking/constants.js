export const OPTION_SEARCH_QUERY_PARAMETERS = 'searchQueryParameters';
export const OPTION_TRANSFORM_PAYLOAD = 'transformPayload';
export const DEFAULT_SEARCH_QUERY_PARAMETERS = ['query', 'searchQuery', 'q'];
export const DEFAULT_CLIENT_LANGUAGE = 'en';
export const CLIENT_LANGUAGES_LIST = [
  DEFAULT_CLIENT_LANGUAGE,
  'ar',
  'da',
  'de',
  'es',
  'fr',
  'it',
  'ja',
  'ko',
  'nl',
  'pt',
  'ru',
  'sv',
  'zh',
];
