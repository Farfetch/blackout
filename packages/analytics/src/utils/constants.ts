import { name, version } from '../../package.json';

export const PACKAGE_VERSION = version;
export const PACKAGE_NAME = name;
export const PACKAGE_NAME_VERSION = `${PACKAGE_NAME}@${PACKAGE_VERSION}`;
export const CONSENT_KEYS = ['marketing', 'statistics', 'preferences'];
