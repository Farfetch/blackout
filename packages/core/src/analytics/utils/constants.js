import { version } from '../../../package.json';

export const PACKAGE_VERSION = version;
export const PACKAGE_NAME = '@farfetch/blackout-core/analytics';
export const PACKAGE_NAME_VERSION = `${PACKAGE_NAME}@${PACKAGE_VERSION}`;
export const DefaultConsentKeys = {
  STATISTICS: 'statistics',
  MARKETING: 'marketing',
  PREFERENCES: 'preferences',
};
export const CONSENT_KEYS = [
  DefaultConsentKeys.STATISTICS,
  DefaultConsentKeys.MARKETING,
  DefaultConsentKeys.PREFERENCES,
];
export const CONSENT_CATEGORIES_PROPERTY = 'consentCategories';
