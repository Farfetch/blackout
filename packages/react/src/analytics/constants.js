// We use a require here to avoid typescript complaining of `package.json` is not
// under rootDir that we would get if we used an import. Typescript apparently ignores
// requires.
// eslint-disable-next-line @typescript-eslint/no-var-requires
const { name, version } = require('../../package.json');

export const PACKAGE_VERSION = version;
export const PACKAGE_NAME = name;
export const PACKAGE_NAME_VERSION = `${PACKAGE_NAME}@${PACKAGE_VERSION}`;
