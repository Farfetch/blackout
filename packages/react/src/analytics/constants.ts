// We use a require here to avoid typescript complaining of `package.json` is not
// under rootDir that we would get if we used an import. Typescript apparently ignores
// requires.
// eslint-disable-next-line @typescript-eslint/no-var-requires
const { name, version } = require('../../package.json');

export const PACKAGE_VERSION = version as string;
export const PACKAGE_NAME = name as string;
export const PACKAGE_NAME_VERSION = `${PACKAGE_NAME}@${PACKAGE_VERSION}`;
