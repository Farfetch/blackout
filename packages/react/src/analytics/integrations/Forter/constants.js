export const ForterTokenReadyEvent = 'ftr:tokenReady';
export const ForterTokenTid = 1364;
export const ForterTokenLoadedAnalyticsEvent = 'forterTokenLoaded';

// site ID for forter sandbox - https://portal.forter.com/api/general/introduction
export const ForterSandboxSiteId = 'b2f674e65110';
// site ID for forter production - https://portal.forter.com/api/general/introduction
export const ForterProductionSiteId = '0c0275c12bca';
