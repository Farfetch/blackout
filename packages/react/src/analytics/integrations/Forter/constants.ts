export const ForterTokenReadyEvent = 'ftr:tokenReady';
export const ForterTokenTid = 1364;
export const ForterTokenLoadedAnalyticsEvent = 'forterTokenLoaded';

export const ForterProductionSiteId = '0c0275c12bca';
