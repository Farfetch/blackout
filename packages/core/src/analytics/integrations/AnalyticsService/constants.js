import { PACKAGE_NAME } from '../../utils/constants';

export const DATA_VERSION_FIELD = '1.0.0';
export const DATA_TYPE_FIELD = PACKAGE_NAME;

export const REQUEST_INTERVAL_KEY = 'requestInterval';
export const MINIMUM_REQUEST_INTERVAL = 1000;
export const DEFAULT_REQUEST_INTERVAL = 5000;
