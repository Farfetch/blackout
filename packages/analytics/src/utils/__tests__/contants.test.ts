import * as constants from '../constants';

describe('constants', () => {
  it('Should export PACKAGE_NAME', () => {
    expect(constants.PACKAGE_NAME).toMatchSnapshot();
  });

  it('Should export CONSENT_KEYS', () => {
    expect(constants.CONSENT_KEYS).toMatchSnapshot();
  });

  it('Should export DefaultConsentKeys', () => {
    expect(constants.DefaultConsentKeys).toMatchSnapshot();
  });

  it('Should export CONSENT_CATEGORIES_PROPERTY', () => {
    expect(constants.CONSENT_CATEGORIES_PROPERTY).toBe('consentCategories');
  });
});
