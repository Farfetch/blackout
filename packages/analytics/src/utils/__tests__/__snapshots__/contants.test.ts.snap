// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`constants Should export CONSENT_KEYS 1`] = `
Array [
  "statistics",
  "marketing",
  "preferences",
]
`;

exports[`constants Should export DefaultConsentKeys 1`] = `
Object {
  "Marketing": "marketing",
  "Preferences": "preferences",
  "Statistics": "statistics",
}
`;

exports[`constants Should export PACKAGE_NAME 1`] = `"@farfetch/blackout-analytics"`;
